src/core/mac/mac.cpp

```diff
@@ -852,6 +919,10 @@ otError Mac::PrepareDataRequest(TxFrame &aFrame)
     aFrame.SetDstAddr(dst);
     IgnoreError(aFrame.SetCommandId(Frame::kMacCmdDataRequest));
 
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    IgnoreError(AppendHeaderIe(false, aFrame));
+#endif
```

Comment:
```suggestion
#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
    IgnoreError(AppendHeaderIe(false, aFrame));
#endif

    IgnoreError(aFrame.SetCommandId(Frame::kMacCmdDataRequest));
```

Otherwise, `SetCommandId` will get wrong payload index thus write command ID to wrong location.